fftools/ffmpeg_enc.c

```diff
@@ -700,7 +700,8 @@ static int encode_frame(OutputFile *of, OutputStream *ost, AVFrame *frame)
             fprintf(ost->logfile, "%s", enc->stats_out);
 
         if (ret == AVERROR(EAGAIN)) {
-            av_assert0(frame); // should never happen during flushing
+            if (!frame)
```

Comment:
I am concerned about this change. According to API description, avcodec_receive_packet() means "output is not available in the current state - user must try to send input". We cannot call avcodec_receive_packet() again.

Reply:
API description only defines user must try to send new input if EAGAIN, but not clear if there is no input left. The main reason I change logic here instead doing the loop inside vaapi encoder is that change here may also benefit other AV1 encoder which has similar process with VAAPI av1 encoder.

Reply:
This means doc/example/vaapi_encode.c and vaapi_transcode.c need to be changed as well.

Reply:
Lots of APP in doc will break and return if get EAGAIN in flushing encoder which will lose some encodes frames at the end of the streams. So change loop inside the vaapi encoder in the new version. Thanks @wenbinc-Bin .